src/provider/view/containers/ToastApp.tsx

```diff
@@ -59,11 +59,17 @@ export function renderApp(
     store: ServiceStore,
     setWindowSize: (dim: WindowDimensions) => void
 ) {
+    const titledNotification: TitledNotification = {
+        ...notification,
+        title: (store.state.applications.get(notification.source.uuid) || {title: notification.source.name || ''}).title
```

Comment:
Is there any nice way to avoid the duplication of this logic between here and `NotificationView`?

Reply:
I want to keep TitledNotification within the UI scope as it's really not relevant to provider logic but a UI representation of the data. I don't think there's a direct common denominator for these two modules at this level. could create a util function but it looks quite simple for that?

Reply:
Yeah that's fair